src/BaGet.UI/src/SearchResults.tsx

```diff
@@ -156,48 +156,9 @@ class SearchResults extends React.Component<ISearchResultsProps, ISearchResultsS
             />
           </div>
         </form>
-        {this.state.items.map(value => (
-          <div key={value.id} className="row search-result">
-            <div className="col-sm-1 hidden-xs hidden-sm">
-              <img
-                src={value.iconUrl || DefaultPackageIcon}
-                className="package-icon img-responsive"
-                onError={this.loadDefaultIcon}
-                alt="The package icon" />
-            </div>
-            <div className="col-sm-11">
-              <div>
-                <Link to={`/packages/${value.id}`} className="package-title">{value.id}</Link>
-                <span>by: {value.authors}</span>
-              </div>
-              <ul className="info">
-                <li>
-                  <span>
-                    <Icon iconName="Download" className="ms-Icon" />
-                    {value.totalDownloads.toLocaleString()} total downloads
-                  </span>
-                </li>
-                <li>
-                  <span>
-                    <Icon iconName="Flag" className="ms-Icon" />
-                    Latest version: {value.version}
-                  </span>
-                </li>
-                {value.tags.length > 0 &&
-                  <li>
-                    <span className="tags">
-                      <Icon iconName="Tag" className="ms-Icon" />
-                      {value.tags.join(' ')}
-                    </span>
-                  </li>
-                }
-              </ul>
-              <div>
-                {value.description}
-              </div>
-            </div>
-          </div>
-        ))}
+
+        {this._renderSearch()}
```

Comment:
I'd rather avoid splitting up the HTML here if possible. Is it possible to do something like:

```tsx
{(() => {
  if (this.state.items.length > 0) {
    // Show results
  } else {
    // Show empty message
  }
})()}
```

Reply:
Yeah for sure. You mean take the code from `_renderSearch` back to the render and do the if inside it. Right?
```suggestion
      render() {
         ...
         <div className="form-group">
            <Checkbox
              defaultChecked={this.state.includePrerelease}
              onChange={this.onChangePrerelease}
              label="Include prerelease:"
              boxSide="end"
            />
          </div>
        </form>
        
        // {this._renderSearch()}
        if (this.state.items.length > 0) {
            this.state.items.map( i => ...
         } else { ...
```

Reply:
To be easier to understand. Something like this? https://gist.github.com/sipmann/9e266ca45333238ccfd9c14f033e2fa2

Reply:
Yup, that looks perfect